core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -86,24 +94,53 @@ public InProcessTelemetryChannel() {
      * @param developerMode True will behave in a 'non-production' mode to ease the debugging
      */
     public InProcessTelemetryChannel(String endpointAddress, boolean developerMode) {
-        initialize(endpointAddress, developerMode);
+        initialize(endpointAddress,
+                   null,
+                   developerMode,
+                   createDefaultMaxTelemetryBufferCapacityEnforcer(null),
+                   createDefaultSendIntervalInSecondsEnforcer(null));
+    }
+
+    /**
+     * Ctor
+     * @param endpointAddress Must be empty string or a valid uri, else an exception will be thrown
+     * @param developerMode True will behave in a 'non-production' mode to ease the debugging
+     * @param maxTelemetryBufferCapacity Max number of Telemetries we keep in the buffer, when reached we will send the buffer
+     *                          Note, value should be between TRANSMIT_BUFFER_MIN_TIMEOUT_IN_MILLIS and TRANSMIT_BUFFER_MAX_TIMEOUT_IN_MILLIS inclusive
+     * @param sendIntervalInMillis The maximum number of milliseconds to wait before we send the buffer
+     *                          Note, value should be between MIN_MAX_TELEMETRY_BUFFER_CAPACITY and MAX_MAX_TELEMETRY_BUFFER_CAPACITY inclusive
+     */
+    public InProcessTelemetryChannel(String endpointAddress, boolean developerMode, int maxTelemetryBufferCapacity, int sendIntervalInMillis) {
+        initialize(endpointAddress,
+                   null,
+                   developerMode,
+                   createDefaultMaxTelemetryBufferCapacityEnforcer(maxTelemetryBufferCapacity),
+                   createDefaultSendIntervalInSecondsEnforcer(sendIntervalInMillis));
     }
 
     /**
      * This Ctor will query the 'namesAndValues' map for data to initialize itself
      * It will ignore data that is not of its interest, this Ctor is useful for building an instance from configuration
-     * @param nameAndValues - The data passed as name and value pairs
+     * @param namesAndValues - The data passed as name and value pairs
      */
-    public InProcessTelemetryChannel(Map<String, String> nameAndValues) {
+    public InProcessTelemetryChannel(Map<String, String> namesAndValues) {
```

Comment:
Why not just using java `Properties` object instead of a map?
Also, `namesAndValues` is a too generic name to be used in the static interface - what are these namesAndValues used for? Where someone can find a documentation about these?